src/Conf.ml

```diff
@@ -836,6 +836,17 @@ let print_config =
   let default = false in
   mk ~default Arg.(value & flag & info ["print-config"] ~doc ~docs)
 
+let root =
+  let docv = "DIR" in
+  let doc =
+    "Root of the project. If specified, only take into account \
+     .ocamlformat configuration files inside $(docv) and its \
+     subdirectories."
+  in
+  let default = None in
+  mk ~default
+    Arg.(value & opt (some dir) default & info ["root"] ~doc ~docs ~docv)
```

Comment:
My feeling is that if root is set, it should be interpreted as if disable-outside-project is also set. Or are there other use cases I'm overlooking? Perhaps it would be best to rename disable-outside-project to something like disable-outside-detected-project?

Reply:
Seems like a good idea to me.